### ISSUE TITLE
KubeMQ Plugin

### ISSUE THREAD

**Issue by Yomanz** [DataDog/dd-trace-js#864]
https://kubemq.io/

It uses grpc but the general grpc plugin lacks some information that's passed.

https://github.com/Yomanz/KubeMQ-TS - The NPM module I made
github.com/kubemq-io/kubemq-node/ - Their official NPM module (mines better :^))

A plugin for this would be dope but I'm not sure where to start with one, are there any documentation for creating a plugin (or even a basic plugin to learn from)

**Comment by rochdev** [comment]
Plugin development is not really documented at the moment since we want to polish the process before documenting it.

In the meantime, I'd say a good way to do it if you want to submit a PR would be in this order:

1) Familiarize yourself with [plugins](https://github.com/DataDog/dd-trace-js/tree/9507b4a261fc970dee9e817a1ea9722af054bb5e/packages) in general. Most of them end up using common patterns so it will give you an idea what metadata to grab etc.
2) Familiarize yourself with a message queue plugin specifically since the flow is a bit different compared to request/response. For example there is [rhea](https://github.com/DataDog/dd-trace-js/tree/9507b4a261fc970dee9e817a1ea9722af054bb5e/packages/datadog-plugin-rhea) that is more recent and more complete, or [amqp10](https://github.com/DataDog/dd-trace-js/tree/9507b4a261fc970dee9e817a1ea9722af054bb5e/packages/datadog-plugin-amqp10) that is a bit older but simpler to read.
3) Once you have an MVP that hooks into the module and creates at least one span, open a WIP PR. This will give us visibility if you have any question, and the PR default description will also guide you in what you need to add (i.e. API docs, TypeScript definitions, etc).
4) At that point it will just be a matter of getting it to the finish line.

In short, familiarize yourself with existing plugins and then open a PR as soon as you have something that works and then we can help :)

**Comment by rochdev** [comment]
A bit unrelated, but it looks like your module has more traction than the official one, and the official one is not GA yet. Did you try to have yours become the official one?

**Comment by Yomanz** [comment]
> 
> 
> A bit unrelated, but it looks like your module has more traction than the official one, and the official one is not GA yet. Did you try to have yours become the official one?

They have an official one but it's coded in NodeJS and there are no types and it was easier, imo, to just create a new one that's in TS and has a generated schema for the grpc endpoint initially ;)

**Comment by rochdev** [comment]
What I'm proposing is to make yours the official one and deprecate the current official one since your library has more traction :)

**Comment by Yomanz** [comment]
I messaged them about it a while back on slack, I'll follow up with them.
